examples/schedulers/static_thread_pool.hpp

```diff
@@ -88,11 +88,229 @@ namespace example {
+            : pool_(pool)
+            , receiver_{(Receiver&&)receiver}
+            , shape_{shape}
+            , fn_{fn}
+            , thread_with_exception_{num_agents_required()}
+          {
+            this->__execute = [](task_base* t, std::uint32_t tid) noexcept {
+              auto& self = *static_cast<bulk_shared_state*>(t);
+              auto total_threads = self.num_agents_required();
+
+              try {
+                std::visit([&](auto& tupl) -> void {
+                  std::apply([&](auto&... args) -> void {
+                    auto [begin, end] = even_share(self.shape_, tid, total_threads);
+                    for (Shape i = begin; i < end; ++i) {
+                      self.fn_(i, args...);
+                    }
+                  }, tupl);
+                }, self.data_);
+              } catch(...) {
+                std::uint32_t expected = total_threads;
+
+                if (self.thread_with_exception_.compare_exchange_strong(
+                        expected, tid, 
+                        std::memory_order_relaxed,
+                        std::memory_order_relaxed)) {
+                  self.exception_ = std::current_exception();
+                }
+              }
+
+              const bool is_last_thread = self.finished_threads_.fetch_add(1) == (total_threads - 1);
+
+              if (is_last_thread) {
+                if (self.exception_) {
+                  if (std::execution::tag_invocable<std::execution::set_error_t, Receiver&&, std::exception_ptr>) {
+                    std::execution::set_error((Receiver&&)self.receiver_, self.exception_);
+                  }
+                } else {
+                  std::visit([&](auto& tupl) noexcept -> void {
+                    std::apply([&](auto&... args) noexcept -> void {
+                        std::execution::set_value((Receiver&&)self.receiver_, (decltype(args)&&)args...);
+                    }, tupl);
+                  }, self.data_);
+                }
+              }
+            };
+          }
+        };
+
+      template <class SenderId, class ReceiverId, class Shape, class Fn>
+        struct bulk_receiver {
+          using Sender = std::__t<SenderId>;
+          using Receiver = std::__t<ReceiverId>;
+
+          using shared_state = bulk_shared_state<SenderId, ReceiverId, Shape, Fn>;
+
+          shared_state& shared_state_;
+
+          template <class... As>
+          friend void tag_invoke(std::execution::set_value_t, bulk_receiver&& self, As&&... as) noexcept {
+            shared_state& state = self.shared_state_;
+
+            if (state.shape_) {
+              try {
+                using tuple_t = std::execution::__decayed_tuple<As...>;
+                state.data_.template emplace<tuple_t>((As &&) as...);
+                state.pool_.bulk_enqueue(&state, state.num_agents_required());
+              } catch (...) {
+                std::execution::set_error((Receiver&&)state.receiver_, std::current_exception());
```

Comment:
Likewise here, we can't instantiate the error path if we know that emplacing the values into the tuple cannot possibly throw, and that also needs to be factored into the `completion_signatures` type computation (which I see is _not_ being done below).